api/apps/restful_apis/file2document_api.py

```
@@ -18,39 +18,51 @@
-def _convert_files(file_ids, kb_ids, user_id):
-    """Synchronous worker: add new docs for the given file/kb pairs while preserving existing links.
-
-    Previously this function replaced all existing links with the new ones, which caused
-    multi-select "link to knowledge base" to overwrite previous links. Now it only creates
-    documents for knowledge bases that are not already linked to the file, and leaves
-    existing links untouched.
-    """
+def _convert_files(file_ids, kb_ids, user_id, mode):
+    """Synchronous worker: add missing links or replace existing links."""
+    replace_existing = mode == "replace"
     for id in file_ids:
```

Ruff (0.15.21)

[error] Variable `id` is shadowing a Python builtin

(A001)